includes/promoted-jobs/class-wp-job-manager-promoted-jobs-api.php

```diff
@@ -174,29 +194,49 @@ public function update_job_status( $request ) {
 	}
 
 	/**
-	 * Get the job data, but only if the user has permission to manage the job and the token is valid.
+	 * Get the job data or error if the job is not found.
 	 *
 	 * @param WP_REST_Request $request Full data about the request.
+	 *
 	 * @return WP_REST_Response|WP_Error The response, or WP_Error on failure.
 	 */
 	public function get_job_data( $request ) {
-		$token    = $request->get_param( 'token' );
-		$user_id  = $request->get_param( 'user_id' );
-		$job_id   = $request->get_param( 'job_id' );
-		$verified = WP_Job_Manager_Site_Trust_Token::instance()->validate( 'user', $user_id, $token );
-		$result   = [
-			'verified' => $verified,
-		];
-		if ( $verified ) {
-			// We only want to return the job data if the specified user has permission to manage the job.
-			if ( 'job_listing' !== get_post_type( $job_id ) ) {
-				return new WP_Error( __( 'Job not found.', 'wp-job-manager' ), [ 'status' => 404 ] );
-			}
-			if ( ! user_can( $user_id, 'manage_job_listings', $job_id ) ) {
-				return new WP_Error( __( 'User does not have enough permissions to get data for this job.', 'wp-job-manager' ), [ 'status' => 401 ] );
+		$job_id = $request->get_param( 'job_id' );
```

Comment:
Oops! The logged-in user check wouldn't work because it's a server request. Another thing I'm thinking of is maybe to mark the job as "open" with a meta key when they click on Promote (before redirecting to wpjobmanager.com). And add a warning (the warning could be part of [this issue](https://github.com/Automattic/WP-Job-Manager/issues/2490)) saying that after proceeding, their job will be exposed through APIs, so other sites can consume it.

Reply:
You are raising a valid point. But I would like to challenge a bit the first assumption: is this a known problem? I mean, are there any use cases we know about that would make the jobs private?

It makes me think about what happens with the default rest API. How does that work for them?

If this is indeed a problem then I would open a separate ticket to work on this, since I can't think of any easy solution to that. (A part from the warning you are suggesting).

Reply:
@renatho I have created [this issue](https://github.com/Automattic/WP-Job-Manager/issues/2514) to continue the discussion, since I am going to merge this into the feature branch.